### PR TITLE
Add substitution in pvl error messages

### PR DESCRIPTION
**Pull request by mlsteele**
Now you can use variables in error messages.

Like this:
```
{ assert_regex_match: "^%{username_service}$"
, case_insensitive: true
, error: ["CONTENT_FAILURE", "Bad post authored; wanted \"%{username_service}\", got \"%{active_string}\""] },
```

Which means pvl can do this, just like the existing system:

```
$ keybase-dev --run-mode=prod --home=/tmp/zzz9 --standalone id -s ellie
✔ public key fingerprint: F845 257D 4679 6465 3815 FFF6 9117 83FB 4845 E4DD
✖ "ellieaskswhy" on twitter failed: Bad post authored; wanted "ellieaskswhy", got "LEKesselman" (code=202)
```

r? @maxtaco

### CODE REVIEW

**Comment by maxtaco** [comment]
It feels a little hacky to me to special case `active_string`; what do you think?

**Comment by mlsteele** [comment]
I didn't want to allow it in instructions because that would allow feeding unknown content from fetches into the regex engine. That's not necessarily bad but there's no reason to allow it. But allowing it in error messages makes for actually helpful error messages. So my rationale for special casing it is that it comes from a different source than the other variables.

**Comment by maxtaco** [comment]
Maybe it would be slightly more general to assume that you'll have more of these, and then you can pass an "extra_substitions" set that in this case contains `active_string`?

**Comment by mlsteele** [comment]
Done, changed `allowActiveString` to a list of strings called `allowedExtras` which can currently contain `"active_string"`.

File: go/pvl/interp.go [hunk 1]
```diff
@@ -941,7 +941,7 @@ func interpretRegex(g proofContextExt, state scriptState, rdesc regexDescriptor)
 	}
 
 	// Do variable interpolation.
-	prepattern, perr := substitute(rdesc.Template, state, nil)
+	prepattern, perr := substitute(rdesc.Template, state, nil, []string{})
```
Comment (maxtaco) [inline go/pvl/interp.go]: you should be save to just pass in `nil` here

File: go/pvl/interp.go [hunk 2]
```diff
@@ -761,7 +761,7 @@ func stepTransformURL(g proofContextExt, ins *jsonw.Wrapper, state scriptState)
 			"Regex transform did not match: %v", sourceRdesc.Template)
 	}
 
-	newURL, err := substitute(destTemplate, state, match)
+	newURL, err := substitute(destTemplate, state, match, []string{})
```
Comment (maxtaco) [inline go/pvl/interp.go]: and similarly, `nil` is ok

**Comment by mlsteele** [comment]
Oh yeah, `[]string{}` replaced with `nil`. And rebased.

**Pull request merged**